cmd/legacy/main.go

```diff
@@ -176,27 +185,41 @@ func (c converter) convertDir(w work, dryRun bool) error {
 		})
 	}
 
+	// If no blocks were read / remain (e.g. due to corruption), we can skip this directory
+	if len(allBlocks) == 0 {
+		return nil
+	}
+
 	// Sort by timestamp to cover potential out-of-order scenarios
 	sort.Slice(allBlocks, func(i, j int) bool {
 		return allBlocks[i].ts < allBlocks[j].ts
 	})
 
-	metadata, err := goDB.ReadMetadata(filepath.Join(w.path, goDB.MetadataFileName))
+	metadata, err := ReadMetadata(filepath.Join(w.path, MetadataFileName))
 	if err != nil {
-		return fmt.Errorf("failed to read metadata from %s: %w", filepath.Join(w.path, goDB.MetadataFileName), err)
+		return fmt.Errorf("failed to read metadata from %s: %w", filepath.Join(w.path, MetadataFileName), err)
 	}
 	writer := goDB.NewDBWriter(c.dbDir, w.iface, encoders.EncoderTypeLZ4)
 
+	var bulkWorkload []goDB.BulkWorkload
 	for _, block := range allBlocks {
 		blockMetadata, err := metadata.GetBlock(block.ts)
 		if err != nil {
 			return fmt.Errorf("failed to get block metdadata from file set: %w", err)
 		}
 
-		if !dryRun {
-			if _, err = writer.Write(block.data, blockMetadata, block.ts); err != nil {
-				return fmt.Errorf("failed to write flows: %w", err)
-			}
+		bulkWorkload = append(bulkWorkload, goDB.BulkWorkload{
+			FlowMap: block.data,
+			CaptureMeta: goDB.CaptureMetadata{
```

Comment:
See comment in its definition:
```
// TODO: This will be replaced by global Capture Stats when merging #47
```
-> Just a blast from the past I kept to minimize conflicts...